powerauth-java-server/src/test/java/io/getlime/security/powerauth/app/server/controller/api/PowerAuthControllerTest.java

```diff
@@ -17,112 +17,1047 @@
+        assertEquals(5, activationListForUserResponse1.getActivations().size());
+
+        final GetActivationListForUserRequest requestPage2 = new GetActivationListForUserRequest();
+        requestPage2.setUserId(baseRequest.getUserId());
+        requestPage2.setApplicationId(baseRequest.getApplicationId());
+        requestPage2.setPageNumber(1);
+        requestPage2.setPageSize(5);
+
+        final GetActivationListForUserResponse activationListForUserResponse2 = powerAuthClient
+                .getActivationListForUser(requestPage2);
+        assertEquals(5, activationListForUserResponse2.getActivations().size());
+
+        assertNotEquals(activationListForUserResponse2.getActivations(), activationListForUserResponse1.getActivations());
+
+        for (final String id : activationIds) {
+            final RemoveActivationResponse removeActivationResponse = powerAuthClient.removeActivation(id, PowerAuthControllerTestConfig.USER_ID);
+            assertTrue(removeActivationResponse.isRemoved());
+        }
+    }
 
+    /**
+     * Tests the activation lookup process based on specific criteria.
+     *
+     * <p>This test executes the following actions:</p>
+     * <ol>
+     *   <li>Constructs a request to look up activations, specifying criteria such as user IDs,
+     *       application IDs, activation status, and a timestamp. The timestamp is set to 10 seconds
+     *       before the current time to include recent activations.</li>
+     *   <li>Sends the lookup request and verifies the response to ensure it contains the expected
+     *       number of activations meeting the specified criteria.</li>
+     *   <li>Asserts that the number of activations in the response matches the expected count,
+     *       confirming the correct functionality of the lookup process.</li>
+     * </ol>
+     *
+     * @throws Exception if any errors occur during the execution of the test or if the assertions fail.
+     */
+    @Test
+    void testLookupActivations() throws Exception {
+        config.initActivation(powerAuthClient);
+        final LookupActivationsRequest lookupActivationsRequest = new LookupActivationsRequest();
+        /* We are looking for an activation created during initialization of the test suite. */
+        final Date timestampCreated = Date.from(LocalDateTime.now().minusSeconds(1).atZone(ZoneId.systemDefault()).toInstant());
+        lookupActivationsRequest.setUserIds(List.of(PowerAuthControllerTestConfig.USER_ID));
+        lookupActivationsRequest.setApplicationIds(List.of(config.getApplicationId()));
+        lookupActivationsRequest.setActivationStatus(ActivationStatus.CREATED);
+        lookupActivationsRequest.setTimestampLastUsedAfter(timestampCreated);
+
+        final LookupActivationsResponse lookupActivationsResponse = powerAuthClient.lookupActivations(lookupActivationsRequest);
+        assertEquals(1, lookupActivationsResponse.getActivations().size());
```

Comment:
FYI: We do not use Hamcrest yet or not heavily, but there are more elegant ways, e.g. `assertThat(collection, hasSize(1));` or some similar syntax with AssertJ (introducing AssertJ is a bigger step, but Hamcrest should be ok, if you want to).